File: backend/models/wyrmspell.py
from enum import Enum

from pydantic import BaseModel

from backend.models.character import Quality


class WyrmspellType(str, Enum):
    BREACH = "Breach"
    REFUGE = "Refuge"
    WILDCRY = "Wildcry"
    DRAGONS_CALL = "Dragon's Call"


class Wyrmspell(BaseModel):
    name: str
    effect: str
    type: WyrmspellType
    quality: Quality
    exclusive_faction: str | None = None
    is_global: bool = True
    last_updated: int | None = None

File: frontend/src/types/wyrmspell.ts
import type { FactionName } from './faction';
import type { Quality } from './quality';

export type WyrmspellType = 'Breach' | 'Refuge' | 'Wildcry' | "Dragon's Call";

export interface Wyrmspell {
  name: string;
  effect: string;
  type: WyrmspellType;
  quality: Quality;
  exclusive_faction: FactionName | null;
  is_global: boolean;
  last_updated: number;
}
